src/Persistence/PersistenceManager.php

```diff
@@ -165,26 +172,45 @@ public function refresh(object &$object, bool $force = false): object
 
         $om = $strategy->objectManagerFor($object::class);
 
-        if ($strategy->contains($object)) {
-            try {
-                $om->refresh($object);
-            } catch (\LogicException|\Error) {
-                // prevent entities/documents with readonly properties to create an error
-                // LogicException is for ORM / Error is for ODM
-                // @see https://github.com/doctrine/orm/issues/9505
-            }
-
-            return $object;
-        }
-
         if ($strategy->isEmbeddable($object)) {
             return $object;
         }
 
-        $id = $om->getClassMetadata($object::class)->getIdentifierValues($object);
+        if (!$strategy->contains($object)) {
```

Comment:
I had to refactor how the refresh was done, this fixes two things:
- if the object is not currently manage by the orm/odm, we need to perform a "merge"
- some stuff had to be made to handle deleted objects. I also manually set to null ids fields that are nullable. Not sure that's the right approach :shrug: